### DISCUSSION TITLE
Recommended design for accessing context/session information in UDF implementation

### DISCUSSION THREAD

**Discussion by sunng87** [apache/datafusion#8871] [Q&A] [answered]
As for now, most datafusion udfs are focusing on data. When I'm going to implement some functions like `database()` or `version()`, I just realized there is no simple way to access information of current session.

For example, in MySQL `SELECT database()` returns current selected database. If we implement `database` as a UDF, we will need to access current session on calling `ScalarFunctionImplementation`. However, the function signature only accepts input data.

Another use case is to access time-zone or locale settings from current session when implementing a date/time related function.

While the function system is under a major refactoring, you may want to add this feature into consideration. Thank you!

**Comment by alamb** [comment, marked as answer]
Hi @sunng87  -- I think the built in `now()` function does something similar -- specifically it needs to know the session start time from the session context. 

In particular it gets access to https://docs.rs/datafusion/latest/datafusion/execution/context/struct.ExecutionProps.html

I wonder if you could use the `var_providers` mechanism or maybe we need to plumb the ExecutionProps into UDFs somehow

**Reply by sunng87** [reply]
Thank you, @alamb ! I will look into `ExecutionProps` and will provide feedback if there is any issue.
